Doc/library/struct.rst

```diff
@@ -194,7 +194,7 @@ platform-dependent.
 +--------+--------------------------+--------------------+----------------+------------+
 | Format | C Type                   | Python type        | Standard size  | Notes      |
 +========+==========================+====================+================+============+
-| ``x``  | pad byte                 | no value           |                |            |
+| ``x``  | pad byte                 | no value           |                | \(7)       |
```

Comment:
It would be best if this footnote were the first.  Is \(0) allowed or must footnote numbering start with 1?
If the latter, I'm NOT asking for renumbering in this situation.

A deeper question: by 'null byte', I presume the footnote means \0.  Why put this in a footnote?  Why not
```suggestion
| ``x``  | \0 pad byte              | no value           |                |            |
```

Reply:
Alternative
```suggestion
| ``x``  | pad byte (\0)           | no value           |                | \(7)       |
```

Reply:
As mentioned on Discord, this isn't a footnote at all, its just arbitrary text—so it could be made anything at all, including 0. They _should_ be made actual footnotes ([1]_/.. [1]), or much better auto-numbered footnotes ([#note-x]_/.. [#note-x]) that would avoid this problem and also getting out of sync and the reader doesn't have to manually scroll multiple pages to find the note (which I've had to do quite a bit on this particular table). However, I considered that out of scope of this particular PR, and better done in a followup.

I presume the clarification in the footnote is for when this is used to output rather than input a binary struct. Its a bit less explicit to inline it, but probably isn't _that_ bad. However, it should be a literal, so

```suggestion
| ``x``  | ``\0`` pad byte          | no value           |                |            |
```

Reply:
What I mean is if you call `struct.unpack` with `x` in the `format` on some struct-packed bytes as input, the bytes ignored by `x` don't have to be `\0` (as the previous suggestion could imply), but rather could be anything. `\0` is only what is output/packed, not nessesarily what is input/unpacked. I.e. on output/packing,

```python
>>> struct.pack('!xxc', b'a')
b'\x00\x00a'
```

but those bytes can be anything on input/unpacking, not just `\0`, i.e.

```python
>>> struct.unpack('!xxc', b'\0\0a')
(b'a',)
>>> struct.unpack('!xxc', b'42a')
(b'a',)
>>> struct.unpack('!xxc', b'\r\na')
(b'a',)
```

where as "\0 pad byte" may imply to the user that only the first might be accepted.